### DISCUSSION TITLE
Building to web and Mac?

### DISCUSSION THREAD

**Discussion by erik-tk** [j20001970/GDMP#22] [Q&A] [answered]
I notice you have no mention of building for Mac or Web. Is there a reason for this or are you not testing these platforms?

**Comment by j20001970** [comment]
Hi, the missing instructions for building library for MacOS is due to lacking the environment for testing . But I think with some adjustment to the build script `build.py`, it is possible to build the dylib without GPU support and load it with Godot, since most components are common across desktops and webcam support is abstracted with OpenCV.

As for web support, last time I checked MediaPipe on web browser are exposed via JavaScript, and GDMP project enables such functionality to GDScript with GDNative libraries. With current situation I'm afraid that supporting for web is out of the project's scope unless MediaPipe source code are capable of compiling into WebAssembly or something. But still there may be alternative solutions to bring MediaPipe into Godot web exports, will investigate the topic more when I got time.

**Comment by j20001970** [comment]
Just found a issue about MediaPipe on WebAssembly that might worth for reference https://github.com/google/mediapipe/issues/877

**Comment by j20001970** [comment, marked as answer]
Hi, GDMP for macOS and Web (only on Godot 4.x) are now available since v0.5, building instructions for macOS have been added in documentations, and instructions on building for Web will be added in near future.

I consider this discussion can be closed, please open separate issue if you encounter problems trying to build or use GDMP on macOS or Web.
